src/elexmodel/models/BaseElectionModel.py

```diff
@@ -264,3 +267,46 @@ def get_coefficients(self):
         These coefficients are for the point prediciton only, not for the lower or upper intervals models.
         """
         return self.features_to_coefficients
+
+    def compute_lambda(
+        self,
+        reporting_units,
+        possible_lambda_values=[],
+        estimand="",
+        K=3,
+    ):
+        if (
+            len(self.features) == 0
+            or len(possible_lambda_values) == 0
+            or estimand == ""
+            or not (f"{estimand}" in reporting_units)
+        ):
+            return 0, 0
+
+        MAPE_arr = np.full_like(possible_lambda_values, 0)  # array of MAPE sums for each lambda
+        kfold = KFold(n_splits=K, shuffle=False)
+
+        # get the data section indexes that we will be training/testing on
+        divisor = 0
+
+        for train_index, test_index in kfold.split(reporting_units):
+            divisor += 1
+            train = reporting_units.iloc[train_index]
+            test = reporting_units.iloc[test_index]
+            # loop through each lambda
+            index = 0
+            for lam in possible_lambda_values:
+                # build model with custom lambda
+                self.lambda_ = lam
+                unit_predictions = self.get_unit_predictions(train, test, estimand)
+                MAPE = mean_absolute_percentage_error(test[estimand].values, unit_predictions.values)
+                MAPE_arr[index] += MAPE
+                index += 1
+
+        # determine average and best
+        MAPE_arr_avg = [value / divisor for value in MAPE_arr]
```

Comment:
You don't need to iterate over the values in a numpy array, you can divide like this:
```
MAPE_arr_avg = MAPE_arr / divisor
```
This does the iteration within numpy, which is often super optimized.